### ISSUE TITLE
Support adding tags for all keywords using `Keyword Tags` setting

### ISSUE THREAD

**Issue by pekkaklarck** [robotframework/robotframework#4373]
Currently it is possible to give tags to user keywords only so that all keywords use the `[Tags]` setting separately. Often all keywords in a resource files have some common tags, and it would be convenient to be able to list them only once. We can support that by adding new `Keyword Tags` setting that can be used in the Settings section. Although the primary use case would we with keywords in resource files, the new setting should work also in test case and suite initialization files. When used in the latter, it should not recurse to lower levels suites or to resource files in the same directory.

With tests it's possible to use `Force Tags` and `Default Tags` setting, but there's a proposal to rename the former to `Test Tags` and removing the latter altogether (#4365). That proposal is related to this issue in two ways:

- If we add `Keyword Tags`, it would be a _lot_ better if the test related setting was `Test Tags` and not `Force Tags`.
- Regardless do we remove `Default Tags` or not, I don't think keyword tags would need that kind of a separate setting. I instead believe we should come up with some new syntax that allows removing tags set globally both from tests and keywords. I'll submit a separate issue about that.

**Comment by oboehmer** [comment]
so the scope of `Keyword Tags` would be all the keywords defined in the file the setting is specified in? I think this is what you mean, @pekkaklarck and what would make sense..

**Comment by pekkaklarck** [comment]
User keywords already can have `[Tags]`. This planned new setting just makes it possible to specify tags for all keywords in a single resource file or a test case file. Instead of
```robotframework
*** Keywords ***
K1
    [Tags]    common
    K
K2
    [Tags]    common
    K
K3
    [Tags]    common    custom
    K
```
you could have
```robotframework
*** Settings ****
Keyword Tags    common

*** Keywords ***
K1
    K
K2
    K
K3
    [Tags]    custom
    K
```
